packages/@expo/dev-server/src/LaunchBrowser.ts

```diff
@@ -0,0 +1,268 @@
+import * as osascript from '@expo/osascript';
+import spawnAsync from '@expo/spawn-async';
+import { spawn, type ChildProcess } from 'child_process';
+import { sync as globSync } from 'glob';
+import open from 'open';
+import os from 'os';
+import path from 'path';
+
+const IS_WSL = require('is-wsl') && !require('is-docker')();
+
+export interface BrowserInstance {
+  close: () => Promise<void>;
+}
+
+/**
+ * Supported browser types
+ */
+enum BrowserTypes {
+  CHROME,
+  EDGE,
+}
+
+/**
+ * Internal browser implementation constraints
+ */
+interface BrowserImpl {
+  /**
+   * Return whether the given `browserType` is supported
+   */
+  isSupportedBrowser: (browserType: BrowserTypes) => Promise<boolean>;
+
+  /**
+   * Launch the browser
+   */
+  launchAsync: (browserType: BrowserTypes, args: string[]) => Promise<BrowserInstance>;
+
+  /**
+   * Close current browser instance
+   */
+  close: () => Promise<void>;
+}
+
+/**
+ * Launch a browser for JavaScript inspector
+ */
+export async function launchBrowserAsync(url: string): Promise<BrowserInstance> {
+  // For dev-client connecting metro in LAN, the request to fetch sourcemaps may be blocked by Chromium
```

Comment:
Would be cool to support `https` in Expo Go, I could add support to the CLI.

Reply:
the request is to metro server, even we could support https, it would possibly be self-signed certificate, so we still need `--allow-running-insecure-content` here.